docs/source/developer/llvm_timings.rst

```diff
@@ -0,0 +1,105 @@
+.. _developer-llvm-timings:
+
+====================
+Notes on timing LLVM
+====================
+
+
+Getting LLVM Pass Timings
+-------------------------
+
+The dispatcher stores LLVM pass timings in the metadata under the
+``llvm_pass_timings`` key. The timings info contains details on how much time
+has spent for each pass. The pass timings are also grouped by their purpose.
+For example, there will be pass timings for function-level pre-optimizations,
+module-level optimizations, and object code generation.
+
+
+Code Example
+~~~~~~~~~~~~
+
+.. literalinclude:: ../../../numba/tests/doc_examples/test_llvm_pass_timings.py
+   :language: python
+   :caption: from ``test_pass_timings`` of ``numba/tests/doc_examples/test_llvm_pass_timings.py``
+   :start-after: magictoken.ex_llvm_pass_timings.begin
+   :end-before: magictoken.ex_llvm_pass_timings.end
+   :dedent: 12
+   :linenos:
+
+Example output:
+
+.. code-block:: text
+
+    Printing pass timings for JITCodeLibrary('DocsLLVMPassTimings.test_pass_timings.<locals>.foo')
```

Comment:
It feels like this section would benefit from some explanation for what the output is showing. I think `#0, #1, #2` are for function passes running on the function, cpython wrapper and cfunc respectively (aside: why is cfunc on by default?). Then `#3` and `#4` are module level passes for the cheap and full runs, but I don't think this split compilation is explained anywhere in the docs? `#5` is finalization which makes sense!

Reply:
I don't think we need too much explanation here. It is reflecting what is happening and I expect things will change as we refine the optimizations later. I'd rather add more descriptive names for each pass group.